sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java

```diff
@@ -0,0 +1,244 @@
+     */
+    public Certificate getCertificate(String alias) {
+        LOGGER.entering("KeyVaultClient", "getCertificate", alias);
+        LOGGER.log(INFO, "Getting certificate for alias: {0}", alias);
+        X509Certificate certificate = null;
+        CertificateBundle certificateBundle = getCertificateBundle(alias);
+        if (certificateBundle != null) {
+            String certificateString = certificateBundle.getCer();
+            if (certificateString != null) {
+                try {
+                    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+                    certificate = (X509Certificate) cf.generateCertificate(
+                        new ByteArrayInputStream(Base64.getDecoder().decode(certificateBundle.getCer()))
+                    );
+                } catch (CertificateException ce) {
+                    LOGGER.log(WARNING, "Certificate error", ce);
+                }
+            }
+        }
+        LOGGER.exiting("KeyVaultClient", "getCertificate", certificate);
+        return certificate;
+    }
+
+    /**
+     * Get the key.
+     *
+     * @param alias the alias.
+     * @param password the password.
+     * @return the key.
+     */
+    public Key getKey(String alias, char[] password) {
+        LOGGER.entering("KeyVaultClient", "getKey", new Object[] { alias, password });
+        LOGGER.log(INFO, "Getting key for alias: {0}", alias);
+        Key key = null;
+        CertificateBundle certificateBundle = getCertificateBundle(alias);
+        boolean isExportable = Optional.ofNullable(certificateBundle)
+                                       .map(CertificateBundle::getPolicy)
+                                       .map(CertificatePolicy::getKey_props)
+                                       .map(KeyProperties::isExportable)
+                                       .orElse(false);
+        if (isExportable) {
+            // Because the certificate is exportable the private key is
+            // available. So we'll use the KeyVault Secrets API to get the 
+            // private key.
+            String certificateSecretUri = certificateBundle.getSid();
+            HashMap<String, String> headers = new HashMap<>();
+            headers.put("Authorization", "Bearer " + getAccessToken());
+            String body = get(certificateSecretUri + API_VERSION_POSTFIX, headers);
+            if (body != null) {
+                JsonConverter converter = JsonConverterFactory.createJsonConverter();
+                SecretBundle secretBundle = (SecretBundle) converter.fromJson(body, SecretBundle.class);
+                try {
+                    KeyStore keyStore = KeyStore.getInstance("PKCS12");
+                    keyStore.load(
+                        new ByteArrayInputStream(Base64.getDecoder().decode(secretBundle.getValue())),
+                        "".toCharArray()
+                    );
+                    alias = keyStore.aliases().nextElement();
```

Comment:
In this particular case because we create new Keystore and we load a specifically with just one certificate `NoSuchElementException' would not be thrown, because if loading failed it would have triggered the catch block.